tools/pre_ci.py

```
@@ -0,0 +1,324 @@
+
+        except subprocess.CalledProcessError as e:
+            print(f"\n❌ FATAL: '{description}' failed with exit code {e.returncode}")
+            if e.stdout:
+                print("--- STDOUT ---")
+                sys.stdout.write(e.stdout)
+            if e.stderr:
+                print("--- STDERR ---")
+                sys.stderr.write(e.stderr)
+
+            # Attach captured outputs to the failure record
+            self.record_result(description, False)
```

Ruff (0.15.11)

[warning] Boolean positional value in function call

(FBT003)